python/ray/data/tests/test_iceberg.py

```diff
@@ -0,0 +1,176 @@
+
+    monkeypatch.setattr(pyi_catalog, "load_catalog", pyi_get_catalog_mock)
+    # monkeypatch.setattr(DataScan, "plan_files", lambda self: list(plan_files) * 10)
+
+
+class TestReadIceberg:
+    def test_get_catalog(self):
+        iceberg_ds = IcebergDatasource(
+            table_identifier=f"{_DB_NAME}.{_TABLE_NAME}",
+            catalog_kwargs={"name": _CATALOG_NAME},
+        )
+        catalog = iceberg_ds._get_catalog()
+        assert catalog.name == _CATALOG_NAME
+
+    def test_plan_files(self):
+        iceberg_ds = IcebergDatasource(
+            table_identifier=f"{_DB_NAME}.{_TABLE_NAME}",
+            catalog_kwargs={"name": _CATALOG_NAME},
+        )
+        plan_files = iceberg_ds.plan_files
+        assert len(plan_files) == 10
+
+    def test_chunk_plan_files(self):
+        iceberg_ds = IcebergDatasource(
+            table_identifier=f"{_DB_NAME}.{_TABLE_NAME}",
+            catalog_kwargs={"name": _CATALOG_NAME},
+        )
+
+        chunks = iceberg_ds._distribute_tasks_into_equal_chunks(
+            iceberg_ds.plan_files, 5
+        )
+        assert (len(c) == 2 for c in chunks)
+
+        chunks = iceberg_ds._distribute_tasks_into_equal_chunks(
+            iceberg_ds.plan_files, 20
+        )
+        assert (
+            sum(len(c) == 1 for c in chunks) == 10
+            and sum(len(c) == 0 for c in chunks) == 10
+        )
+
+    def test_get_read_tasks(self):
+        iceberg_ds = IcebergDatasource(
+            table_identifier=f"{_DB_NAME}.{_TABLE_NAME}",
+            catalog_kwargs={"name": _CATALOG_NAME, "type": "sql"},
+        )
+        read_tasks = iceberg_ds.get_read_tasks(5)
+        assert len(read_tasks) == 5
+        assert all(len(rt.metadata.input_files) == 2 for rt in read_tasks)
+
+    def test_filtered_read(self):
+        from pyiceberg import expressions as pyi_expr
+
+        iceberg_ds = IcebergDatasource(
+            table_identifier=f"{_DB_NAME}.{_TABLE_NAME}",
+            row_filter=pyi_expr.In("col_c", {1, 2, 3, 4}),
+            selected_fields=("col_b",),
+            catalog_kwargs={"name": _CATALOG_NAME, "type": "sql"},
+        )
+        read_tasks = iceberg_ds.get_read_tasks(5)
+        # Should be capped to 4, as there will be only 4 files
+        assert len(read_tasks) == 4, read_tasks
+        assert all(len(rt.metadata.input_files) == 1 for rt in read_tasks)
+
+
+def test_read_basic(pyiceberg_full_mock):
+    ray_ds = read_iceberg(
+        table_identifier=f"{_DB_NAME}.{_TABLE_NAME}",
+        row_filter=pyi_expr.In("col_c", {1, 2, 3, 4, 5, 6, 7, 8}),
+        selected_fields=(
+            "col_a",
+            "col_b",
+        ),
+        catalog_kwargs={"name": _CATALOG_NAME, "type": "sql"},
+    )
+    table: pa.Table = pa.concat_tables((ray.get(ref) for ref in ray_ds.to_arrow_refs()))
+
+    expected_schema = pa.schema(
+        [pa.field("col_b", pa.int32()), pa.field("col_b", pa.string())]
+    )
+    assert table.schema.equals(expected_schema)
```

Comment:
to trigger pytest to run, we should add something like this at the bottom of the file (i see it [in the CI](https://buildkite.com/ray-project/microcheck/builds/3855#01912d76-71a8-4ecf-ae89-e50e6a95d3cb/185-1032), but i think 2 seconds seems fast, so likely the tests are not actually running):
```
if __name__ == "__main__":
    import sys

    sys.exit(pytest.main(["-v", __file__]))
```